core/src/main/java/io/questdb/griffin/engine/functions/bool/InSymbolVarcharArrayFunctionFactory.java

```diff
@@ -0,0 +1,165 @@
+    ) throws SqlException {
+        SymbolFunction symbolFunc = (SymbolFunction) args.getQuick(0);
+        Function arrayFunc = args.getQuick(1);
+        if (!arrayFunc.isConstant() && !arrayFunc.isRuntimeConstant()) {
+            throw SqlException.$(argPositions.getQuick(1), "constant or bind variable expected");
+        }
+        return new Func(symbolFunc, arrayFunc);
+    }
+
+    private static class Func extends BooleanFunction implements BinaryFunction {
+        private final Function arrayFunc;
+        private final SymbolFunction symbolFunc;
+        private IntHashSet intSet;
+        private boolean stateInherited = false;
+        private CharSequenceHashSet strSet;
+        private boolean useIntSet;
+
+        public Func(SymbolFunction symbolFunc, Function arrayFunc) {
+            this.symbolFunc = symbolFunc;
+            this.arrayFunc = arrayFunc;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            return useIntSet
+                    ? intSet.contains(symbolFunc.getInt(rec))
+                    : strSet.contains(symbolFunc.getSymbol(rec));
+        }
+
+        @Override
+        public Function getLeft() {
+            return symbolFunc;
+        }
+
+        @Override
+        public Function getRight() {
+            return arrayFunc;
+        }
+
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            BinaryFunction.super.init(symbolTableSource, executionContext);
+            if (stateInherited) {
+                return;
+            }
+
+            if (intSet == null) {
+                intSet = new IntHashSet();
+            }
+            if (strSet == null) {
+                strSet = new CharSequenceHashSet();
+            }
+            intSet.clear();
+            strSet.clear();
+            ArrayView arrayView = arrayFunc.getArray(null);
+            StaticSymbolTable symbolTable = symbolFunc.getStaticSymbolTable();
+
+            if (symbolTable != null) {
+                for (int i = 0, n = arrayView.getCardinality(); i < n; i++) {
+                    Utf8Sequence value = arrayView.getVarchar(i);
+                    if (value == null) {
+                        intSet.add(symbolTable.keyOf(null));
+                    } else if (value.isAscii()) {
+                        intSet.add(symbolTable.keyOf(value.asAsciiCharSequence()));
+                    } else {
+                        StringSink sink = Misc.getThreadLocalSink();
+                        sink.put(value);
+                        intSet.add(symbolTable.keyOf(sink));
+                    }
+                }
+                useIntSet = true;
+            } else {
+                for (int i = 0, n = arrayView.getCardinality(); i < n; i++) {
+                    Utf8Sequence element = arrayView.getVarchar(i);
+                    strSet.add(Utf8s.toString(element));
```

Comment:
we should optimise for ascii chars. I suspect we can avoid string creation here altogether